packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart

```diff
@@ -118,276 +119,295 @@ If this option is not provided, the version number will be determined from the p
+              final confirm = logger.confirm('Would you like to continue?');
+
+              if (!confirm) {
+                logger.info('Aborting.');
+                return ExitCode.success.code;
+              }
+            }
+
+            await codePushClientWrapper.publishPatch(
+              appId: appId,
+              releaseId: release.id,
+              metadata: CreatePatchMetadata(
+                releasePlatform: releasePlatform,
+                usedIgnoreAssetChangesFlag: allowAssetDiffs,
+                hasAssetChanges: diffStatus.hasAssetChanges,
+                usedIgnoreNativeChangesFlag: allowNativeDiffs,
+                hasNativeChanges: diffStatus.hasNativeChanges,
+                linkPercentage: null,
+                environment: BuildEnvironmentMetadata(
+                  operatingSystem: platform.operatingSystem,
+                  operatingSystemVersion: platform.operatingSystemVersion,
+                  shorebirdVersion: packageVersion,
+                  xcodeVersion: null,
+                ),
+              ),
+              platform: releasePlatform,
+              track: isStaging
+                  ? DeploymentTrack.staging
+                  : DeploymentTrack.production,
+              patchArtifactBundles: patchArtifactBundles,
+            );
 
-          if (!confirm) {
-            logger.info('Aborting.');
             return ExitCode.success.code;
-          }
-        }
-
-        await codePushClientWrapper.publishPatch(
-          appId: appId,
-          releaseId: release.id,
-          metadata: CreatePatchMetadata(
-            releasePlatform: releasePlatform,
-            usedIgnoreAssetChangesFlag: allowAssetDiffs,
-            hasAssetChanges: diffStatus.hasAssetChanges,
-            usedIgnoreNativeChangesFlag: allowNativeDiffs,
-            hasNativeChanges: diffStatus.hasNativeChanges,
-            linkPercentage: null,
-            environment: BuildEnvironmentMetadata(
-              operatingSystem: platform.operatingSystem,
-              operatingSystemVersion: platform.operatingSystemVersion,
-              shorebirdVersion: packageVersion,
-              xcodeVersion: null,
-            ),
-          ),
-          platform: releasePlatform,
-          track:
-              isStaging ? DeploymentTrack.staging : DeploymentTrack.production,
-          patchArtifactBundles: patchArtifactBundles,
+          },
+          values: {
+            shorebirdEnvRef.overrideWith(() => releaseFlutterShorebirdEnv),
+          },
         );
-
-        return ExitCode.success.code;
-      },
-      values: {
-        shorebirdEnvRef.overrideWith(() => releaseFlutterShorebirdEnv),
-      },
-    );
+      } on ArtifactNotFoundException catch (error) {
+        logger.err(error.toString());
+        return ExitCode.software.code;
+      } on MultipleArtifactsFoundException catch (error) {
+        logger.err(error.toString());
+        return ExitCode.software.code;
+      }
+    });
```

Comment:
I think you may still be working on this, but if this logic stays in the patch command, it should probably be more local to the build commands, even if it results in some duplicated code.

Reply:
yeah, I uploaded this PR, which I guess made patch command working. But I am working in parallel in a better version of it, that implements this as we discussed internally